File: backend/app/models/customer.py
from sqlalchemy import Column, Integer, String, DateTime, Enum
from sqlalchemy.orm import relationship
from datetime import datetime
from app.models.base import Base
import enum

class CustomerType(enum.Enum):
    individual = "individual"  # Pessoa Física
    company = "company"        # Pessoa Jurídica

class Customer(Base):
    __tablename__ = "customers"

    id = Column(Integer, primary_key=True)

    type = Column(Enum(CustomerType), nullable=False, default=CustomerType.individual)

    full_name = Column(String, nullable=False)
    trade_name = Column(String)
    legal_name = Column(String)
    document = Column(String, nullable=False, unique=True, index=True)

    rg = Column(String)
    mother_name = Column(String)
    father_name = Column(String)

    email = Column(String, nullable=False)
    phone = Column(String)
    mobile1 = Column(String)
    mobile2 = Column(String)
    fax = Column(String)

    address_street = Column(String, nullable=False)
    address_number = Column(String, nullable=False)
    address_complement = Column(String)
    address_neighborhood = Column(String, nullable=False)
    address_city = Column(String, nullable=False)
    address_state = Column(String, nullable=False)
    address_zipcode = Column(String, nullable=False)

    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

    services = relationship("CustomerService", back_populates="customer")

File: frontend/src/app/schemas/customer.ts
// schemas/customer.ts
import * as yup from "yup";

export const customerSchema = yup.object().shape({
  type: yup.string().required(),
  full_name: yup.string().required("Nome completo é obrigatório"),
  document: yup.string().required("CPF/CNPJ é obrigatório"),
  email: yup.string().email("Email inválido").required("Email é obrigatório"),
  phone: yup.string(),
  mobile1: yup.string(),
  address_street: yup.string().required("Rua é obrigatória"),
  address_number: yup.string().required("Número é obrigatório"),
  address_complement: yup.string(),
  address_neighborhood: yup.string().required("Bairro é obrigatório"),
  address_city: yup.string().required("Cidade é obrigatório"),
  address_state: yup.string().required("Estado é obrigatório"),
  address_zipcode: yup.string().required("CEP é obrigatório"),
  trade_name: yup.string(),
  legal_name: yup.string(),
  rg: yup.string(),
  mother_name: yup.string(),
  father_name: yup.string(),
});
